generator/address_parser/processor.cpp

```diff
@@ -0,0 +1,93 @@
+#include "processor.hpp"
+
+#include "tiger_parser.hpp"
+
+#include "geometry/mercator.hpp"
+
+#include "coding/point_coding.hpp"
+
+#include "base/file_name_utils.hpp"
+
+#include "defines.hpp"
+
+
+namespace addr_generator
+{
+
+Processor::Processor(std::string const & dataPath, std::string const & outputPath, size_t numThreads)
+  : m_affiliation(dataPath, true /* haveBordersForWholeWorld */)
+  , m_workers(numThreads)
+  , m_outputPath(outputPath)
+{
+}
+
+FileWriter & Processor::GetWriter(std::string const & country)
+{
+  auto res = m_country2writer.try_emplace(country, base::JoinPath(m_outputPath, country) + TEMP_ADDR_EXTENSION);
+  return res.first->second;
+}
+
+void Processor::Run(std::istream & is)
+{
+  std::atomic<size_t> incomplete = 0;
+  size_t total = 0;
+
+  std::string line;
+  while (std::getline(is, line))
+  {
+    ++total;
+
+    m_workers.SubmitWork([this, &incomplete, copy = std::move(line)]() mutable
```

Comment:
How big was the speedup compared to single-thread processing without copying strings and mutex?